packages/pigeon/platform_tests/test_plugin/example/ios/RunnerTests/EchoBinaryMessenger.swift

```diff
@@ -28,13 +28,14 @@ class EchoBinaryMessenger: NSObject, FlutterBinaryMessenger {
     
     guard
       let args = self.codec.decode(message) as? [Any?],
-      let firstArg: Any? = nilOrValue(args.first)
```

Comment:
`args.first` is another nested optional (`Any??`) - outer nil represents "there's no first element, or list is empty", and inner nil represents "there is first element, but it is nil". 

When passing it to `nilOrValue`, it gets coerced into `Any?`, hence the compiler warning in Xcode (Bonus: we should make this an error instead of warning, will do in separate PR).

Reply:
This is very strange, I would expect the old code and the new code to behave in the same way.

Reply:
The old code worked because it has "as Any" intermediate cast inside nilOrValue function. The new code here avoids implicitly optional completely, so no need the hack anymore.